mlflow/pyfunc/__init__.py

```diff
@@ -404,24 +421,29 @@ class PyFuncModel(object):
     ``model_meta`` contains model metadata loaded from the MLmodel file.
     """
 
-    def __init__(self, model_meta: Model, model_impl: Any):
+    def __init__(self, model_meta: Model, model_impl: Any, supports_tensor_input: bool = False):
         if not hasattr(model_impl, "predict"):
             raise MlflowException("Model implementation is missing required predict method.")
         if not model_meta:
             raise MlflowException("Model is missing metadata.")
         self._model_meta = model_meta
         self._model_impl = model_impl
+        self._supports_tensor_input = supports_tensor_input
 
-    def predict(self, data: pandas.DataFrame) -> PyFuncOutput:
+    def predict(self, data: Any) -> PyFuncOutput:
         """
         Generate model predictions.
-        :param data: Model input as pandas.DataFrame.
+        :param data: Model input
         :return: Model predictions as one of pandas.DataFrame, pandas.Series, numpy.ndarray or list.
         """
         input_schema = self._model_meta.get_input_schema()
         if input_schema is not None:
             data = _enforce_schema(data, input_schema)
-        return self._model_impl.predict(data)
+        if isinstance(data, pandas.DataFrame) or (
+            self._supports_tensor_input and _is_supported_tensor(data)
+        ):
+            return self._model_impl.predict(data)
+        raise MlflowException("Provided input data is not of a supported type")
```

Comment:
Can we be more specific when raising exception? 
1. It would be good to know if the input does not work because the model does not support tensor input or the input is not supported altogether. 
2. We should also include the list of supported data types, something like MLflow python function interface supports the following input data: pandas.DataFrame or numpy array for tabular data, numpy array or a list of numpy array or a dictionary of str -> numpy array for tensors. If you are trying to score other data type that is understood by your model, please load your model via appropriate model flavor, e.g. `mlflow.pytorch.load_model`.